### ISSUE TITLE
Accessibility errors triggered by "coreyes" links

### ISSUE THREAD

**Issue by Steve-Noble** [w3c/mathml#407]
Automated accessibility checkers (both WAVE and Accessibility Insights) throw accessibility errors in all of the places where we have a link styled as "core" followed by a check mark.  A related basic underlying principle is that you don't want a bunch of different links on a page which all go to different places but have the same text...like a page which uses "click here" for all the links. Generally, this falls under Success Criterion 2.4.4 Link Purpose (In Context) https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html

As discussed in https://www.w3.org/WAI/WCAG21/Techniques/html/H33, one possible fix without changing the visual styling is to include a title tag inside the anchor element tag. Something like this perhaps: `<a class="coreyes" href="https://w3c.github.io/mathml-core/spec.html#dfn-display" title="display attribute"></a>` [Where the "title=" text gives a specific description of where the link goes, not just the same generic text for all links.]

Of course, there is often some gray area on interpretation of WCAG criteria, and one might argue that a screen reader can figure out where the link will go from the context. But when a screen reader user queries the page for a list of links or navigates by moving to the next link or by tabbing, all of these "core yes" links are going to sound the same. For example, in NVDA they are all read as "core check." So, I think the automated checkers' algorithm for marking this as an accessibility error is most likely valid.

@NSoiffer suggested "For this case, the problem occurs because there is no content -- it is coming from CSS::before. I think whether AT sees this remains a controversy. To side step it, I think using the actual content rather than "::before" is best. Barring that, I think (but need to check) aria-label="core&#x2714;" would work."

FYI: For an aria-label example on anchor elements, see https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8

{screenshot of example below}
![image](https://user-images.githubusercontent.com/11201669/181634708-ce582608-94f8-4222-aa27-d0488bdc9eb1.png)

**Comment by NSoiffer** [comment]
My comment was not appropriate -- I read the issue too quickly. The issue is the link, not the text. The text reads properly in NVDA although the "x" reads as "times".

I manually added a `title`. In Chrome+NVDA, it was _not_ read out, so this is not a useful solution despite the WCAG suggestions... at least not in Chrome+NVDA. It does show as a tooltip, but the tooltip was not read.

**Comment by Steve-Noble** [comment]
I guess this is one of those cases where a particular accessibility technique may be applied which would appease the automated checking tools, even though a screen reader might not take advantage of that method. (Which is why human usability testing is needed as well).

There are many more possible methods of satisfying the criterion, as mentioned under https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html#techniques

Of course, the most reliable method is to include some specific unambiguous text to which the anchor tag is applied.

**Comment by NSoiffer** [comment]
I tried `title` in Firefox+NVDA and it didn't do anything.

I tried `aria-label` and it works in Firefox+NVDA but not Chrome+NVDA. I would have thought that would work everywhere. :-(

**Comment by davidcarlisle** [comment]
> I guess this is one of those cases where a particular accessibility technique may be applied which would appease the automated checking tools, even though a screen reader might not take advantage of that method. (Which is why human usability testing is needed as well).

Another way of seeing the issue is to say it's a case where automated checking can flag errors which human testing shows is not an issue.

In the two headings

**`<mn>` <a href="mathml-core#mn">core</a>**


**`<mi>` <a href="mathml-core#mi">core</a>**

The "core" link is a used as a "badge" for elements in core, it would not serve any purpose to modify the link text and repeat the element name





**`<mn>` <a href="mathml-core#mn">core mn</a>**


**`<mi>` <a href="mathml-core#mi">core mi</a>**



We could move the `core✔` from CSS  `::before` to real content if that would help?

**Comment by Steve-Noble** [comment]
If the real intent is to serve as an icon to let people know it is in core, then I guess treating it the same as "coreno" would not be an issue. It is only when we embed a link at that spot that some further treatment is needed. Besides, the link to the point in the spec is already present in the "values" column on these tables, so also including the link on the badge is not necessary.

By the way, as an aside the badges are reading for me in NVDA as "core check" for yes, and "core multiply" for no. Ideally, we want something more direct...like "yes" and "no"

**Comment by davidcarlisle** [comment]
> By the way, as an aside the badges are reading for me in NVDA as "core check" for yes, and "core multiply" for no. Ideally, we want something more direct...like "yes" and "no"

Ah we should definitely fix that. Could lose the check marks and have `core` and `not core` or fix the cross to use ❌ (U+274C, CROSS MARK) I thought I'd used that but seems I used  U+2716  (HEAVY MULTIPLICATION X) which is clearly wrong, sorry.

**Comment by davidcarlisle** [comment]
I have removed the mis-used check and times symbols, currently "core" still links to the core spec.

> Besides, the link to the point in the spec is already present in the "values" column on these tables, so also including the link on the badge is not necessary.

This is only true for some attributes and not true for elements.

eg

https://w3c.github.io/mathml/spec.html#presm_mroot

the only links to the core versions of msqrt and mroot are the core links.  If the only way to pass mechanical accessibility tests is to remove the link,  we could do that although removing information for everybody seems a sub-optimal way to pass such tests. 

However In the above commit I have changed (just) mroot and msqrt in that heading to have the tag syntax as well as the core badge within the link, is that better?  I don't like it as much as really the `<msqrt>` is a heading for the _full_ version and the <sup>core</sup> was the link to core, but it's OK, and we could do this throughout if it closes this issue?
